toolkit/tools/imagegen/configuration/partitionflag.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Parser for the image builder's configuration schemas.
+
+package configuration
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// PartitionFlag describes the features of a partition
+type PartitionFlag string
+
+const (
+	// PartitionFlagESP indicates this is the UEFI esp partition
+	PartitionFlagESP PartitionFlag = "esp"
+	// PartitionFlagGrub indicates this is a grub boot partition
+	PartitionFlagGrub PartitionFlag = "grub"
+	// PartitionFlagBiosGrub indicates this is a bios grub boot partition
+	PartitionFlagBiosGrub PartitionFlag = "bios_grub"
+	// PartitionFlagBoot indicates this is a boot partition
+	PartitionFlagBoot PartitionFlag = "boot"
+	// PartitionFlagDeviceMapperRoot indicates this partition will be used for a device mapper root device
+	PartitionFlagDeviceMapperRoot PartitionFlag = "dmroot"
+)
+
+func (p PartitionFlag) String() string {
+	return fmt.Sprint(string(p))
+}
+
+// GetValidPartitionFlags returns a list of all the supported
```

Comment:
Nit: does this need to be public if we only needed for validation inside this file (and the test)?

Reply:
There are a few places we use similar functions elsewhere in the code and do access them publicly. Figured I would keep with the pattern I established for the other structs (for better or worse).